pkg/controller/master/setting/auto_disk_provision.go

```diff
@@ -0,0 +1,52 @@
+package setting
+
+import (
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	NDMDaemonSetName        = "harvester-node-disk-manager"
+	NDMContainerName        = "harvester-node-disk-manager"
+	NDMAutoProvisionEnvName = "NDM_AUTO_PROVISION_FILTER"
+)
+
+func (h *Handler) syncNDMAutoProvisionPaths(setting *harvesterv1.Setting) error {
+	ds, err := h.daemonSetCache.Get(util.HarvesterSystemNameSpaceName, NDMDaemonSetName)
+	if err != nil {
+		return err
+	}
+	dsCopy := ds.DeepCopy()
+
+	for i, container := range dsCopy.Spec.Template.Spec.Containers {
+		if container.Name == NDMContainerName {
+			envFound := false
+			for j, env := range container.Env {
+				if env.Name == NDMAutoProvisionEnvName {
+					envFound = true
+					logrus.Infof("update auto provision paths filter to '%s'", setting.Value)
+					dsCopy.Spec.Template.Spec.Containers[i].Env[j].Value = setting.Value
+					break
+				}
+			}
+			if !envFound {
+				logrus.Infof("add auto provision paths filter '%s'", setting.Value)
+				dsCopy.Spec.Template.Spec.Containers[i].Env = append(
+					dsCopy.Spec.Template.Spec.Containers[i].Env,
+					corev1.EnvVar{Name: NDMAutoProvisionEnvName, Value: setting.Value},
+				)
+			}
+
+			break
+		}
+	}
+
+	if _, err := h.daemonSets.Update(dsCopy); err != nil {
```

Comment:
NDM is deployed using helm but we are patching the daemonset directly. My concern is the changes will be lost during upgrades. Should we update via helm release? For example, using https://docs.rke2.io/helm/#customizing-packaged-components-with-helmchartconfig

Reply:
I need to check this out, thanks!

Reply:
@gitlawr unless we also update the file `/var/lib/rancher/rke2/server/manifests/harvester-config.yaml`, everything would be reverted by RKE2 after a reboot. Also, we can't directly modify the file since it's generated by cOS-toolkit's OEM file `/oem/99_custom.yaml`.

Reply:
I found this is a common issue that we have encountered in many places and has raised it on the https://suse.slack.com/archives/C02D01ARDL5/p1636703672048800